src/test/java/uk/gov/ida/verifyserviceprovider/configuration/EuropeanIdentityConfigurationTest.java

```diff
@@ -0,0 +1,87 @@
+        europeanIdentityConfiguration.setEnvironment(HubEnvironment.INTEGRATION);
+        Certificate europeanConfigCert =  europeanIdentityConfiguration.getTrustStore().getCertificate("metadataCA");
+
+        assertThat(europeanIdentityConfiguration.getTrustStore().containsAlias("rootCA")).isTrue();
+        assertThat(europeanIdentityConfiguration.getTrustStore().containsAlias("metadataCA")).isTrue();
+        assertThat(europeanIdentityConfiguration.getTrustStore().size()).isEqualTo(2);
+        assertThat(europeanConfigCert).isEqualTo(testEntryCert);
+    }
+
+    @Test
+    public void shouldUseTrustStoreWithProductionTrustAnchorGivenEidasIsEnabledWithHubEnvironmentSetToProduction() throws Exception {
+
+        Certificate testEntryCert =  keyStore.getKeyStore().getCertificate("metadataCA");
+        EuropeanIdentityConfiguration europeanIdentityConfiguration = OBJECT_MAPPER.readValue(config, EuropeanIdentityConfiguration.class);
+        europeanIdentityConfiguration.setEnvironment(HubEnvironment.PRODUCTION);
+        Certificate europeanConfigCert =  europeanIdentityConfiguration.getTrustStore().getCertificate("metadataCA");
+
+        assertThat(europeanIdentityConfiguration.getTrustStore().containsAlias("rootCA")).isTrue();
+        assertThat(europeanIdentityConfiguration.getTrustStore().containsAlias("metadataCA")).isTrue();
+        assertThat(europeanIdentityConfiguration.getTrustStore().size()).isEqualTo(2);
+        assertThat(europeanConfigCert).isEqualTo(testEntryCert);
+    }
+
+    @Test
+    public void shouldUseTestTrustStoreWithComplianceTrustAnchorGivenEidasIsEnabledWithHubEnvironmentSetToCompliance() throws Exception {
+
+        Certificate testEntryCert =  keyStore.getKeyStore().getCertificate("metadataCA");
+        EuropeanIdentityConfiguration europeanIdentityConfiguration = OBJECT_MAPPER.readValue(config, EuropeanIdentityConfiguration.class);
+        europeanIdentityConfiguration.setEnvironment(HubEnvironment.COMPLIANCE_TOOL);
+        Certificate europeanConfigCert =  europeanIdentityConfiguration.getTrustStore().getCertificate("metadataCA");
+
+        assertThat(europeanIdentityConfiguration.getTrustStore().containsAlias("rootCA")).isTrue();
+        assertThat(europeanIdentityConfiguration.getTrustStore().containsAlias("metadataCA")).isTrue();
+        assertThat(europeanIdentityConfiguration.getTrustStore().size()).isEqualTo(2);
+        assertThat(europeanConfigCert).isEqualTo(testEntryCert);
+    }
+
+    @Test
+    public void shouldReadTrustAnchorGivenEidasIsEnabledWithASpecifiedTrustAnchorUri() throws IOException {
+
+        EuropeanIdentityConfiguration europeanIdentityConfiguration = OBJECT_MAPPER.readValue(config, EuropeanIdentityConfiguration.class);
+        europeanIdentityConfiguration.setEnvironment(HubEnvironment.INTEGRATION);
+
+        assertThat(europeanIdentityConfiguration.getTrustAnchorUri()).isEqualTo(URI.create("http://example.com"));
+    }
```

Comment:
It doesn't feel to me like these tests cover the different trust anchor possibilities - we're setting the trust anchor URI to http://example.com in all cases.  I think the assertion that you've got in shouldReadTrustAnchorGivenEidasIsEnabledWithASpecifiedTrustAnchorUri would still pass in shouldUseTestTrustStoreWithIntegrationTrustAnchorGivenEidasIsEnabledWithHubEnvironmentSetToIntegration.

It might not be easy to make the behaviour different in different tests though, at least if the tests are all in one class. :(

Reply:
Thanks for your comments, I am open to pointers on how to do this test.

Reply:
So I have renamed the test in your comment to
```
shouldReadTrustAnchorAtUriGivenEidasIsEnabledWithoutHubEnvironmentSetToIntegrationWhenTrustAnchorUriHasbeenSpecified```
And added two others
```

- shouldReadHubEnvironmentTrustAnchorGivenEidasIsEnabledWithHubEnvironmentSetToIntegrationWhenNoTrustAnchorUriHasbeenSpecified
- shouldErrrorWhenReadingTrustAnchorGivenEidasIsEnabledWithNoSpecifiedTrustAnchorUriAndNoSpecifiedEnvironment

```
for better test coverage.

Additionally hardcoded Urls have been moved into `ConfigurationConstants` class

Reply:
I don't think these tests really test what they say they do, or am I missing something?